src/USER-REACTION/fix_bond_react.cpp

```diff
@@ -3781,8 +3926,11 @@ void FixBondReact::ReadConstraints(char *line, int myrxn)
         if (ifragment < 0) error->one(FLERR,"Bond/react: Molecule fragment does not exist");
         else constraints[i][myrxn].id[0] = ifragment;
       }
-    } else
-      error->one(FLERR,"Bond/react: Illegal constraint type in 'Constraints' section of map file");
+    } else if (strcmp(constraint_type,"custom") == 0) {
+      constraints[i][myrxn].type = CUSTOM;
+      strtok(line," \t\n\r\f");
+      constraints[i][myrxn].str = strtok(nullptr,"\"\t\n\r\f");
```

Comment:
Please consider using the Tokenizer class instead of strtok() which is not thread safe.

Reply:
I am not sure if this can be done simply with the Tokenizer class: I just need to get the second 'word', which may be enclosed in quotes. If not, I suppose I could do it manually.

Reply:
perhaps `utils::split_words()` can do what you are looking for: https://docs.lammps.org/Developer_utils.html#string-processing

Reply:
perfect, thanks